src/client_side_reply.cc

```diff
@@ -1122,6 +1088,22 @@ clientReplyContext::purgeDoPurgeHead(StoreEntry *newEntry)
     http->storeEntry()->complete();
 }
 
+void
+clientReplyContext::purgeEntry(StoreEntry *entry, const Http::MethodType &methodType, const char *descriptionPrefix)
+{
+    if (!entry) {
+        if (purgeStatus == Http::scNone)
+            purgeStatus = Http::scNotFound;
```

Comment:
AFAICT, this is a PR bug: purgeStatus may become scNotFound prematurely and will stay that way despite matching entries being found later.

In general, I think we should avoid changing purgeStatus to an incorrect status that would have to be changed later. That implies not changing purgeStatus to scNotFound until we know that no matching entries have been found. IIRC, this was discussed earlier, and the current official code follows that principle.

Reply:
> this is a PR bug: purgeStatus may become scNotFound prematurely

Sorry, I don't see a bug here. Initially, purgeStatus is scNone. When clientReplyContext::purgeEntry() is called the first time, it becomes either scNotFound or scOkay. Further calls (if an entry was found) will make it scOkay.
Also, IMO, it is more reliable to keep purgeStatus 'up-to-date' at each step. In other words, we should not rely on some other code, that will probably be called in the future and set it to scNotFound (if nothing was found by then).

Reply:
> Further calls (if an entry was found) will make it scOkay.

Yes, you are right, sorry.

> Also, IMO, it is more reliable to keep purgeStatus 'up-to-date' at each step.

I agree. We disagree what "up to date" means (or what "purgeStatus" is). To me, in the _middle_ of the processing steps, the value scNotFound is not an "up to date" value for purgeStatus because we have not reached that conclusion at any point in time yet. However, I can certainly see the other point of view as well. I prefer avoiding scNotFound status until the end because doing so might help with exposing bugs where the processing sequence got interrupted early.

One way to avoid this problem itself is to replace purgeStatus with a purgedEntries counter. A counter would also carry more information for triage!